lib/kernel/src/erl_epmd.erl

```diff
@@ -378,7 +373,24 @@ get_epmd_port() ->
 	error ->
 	    ?erlang_daemon_port
     end.
-	    
+
+erl_epmd_listen_port() ->
```

Comment:
This change does not affect compatibility, so specifying either should work exactly the same. Or am I missing something else?

Reply:
Yes, specifying either should work, but specifying both of them should crash :)

Reply:
> If version OTP 27.X is not compatible with OTP 27.Y, the user should choose the old one to maintain compatibility.

I think my confusion is because in this case the compatibility between version is kept, so it does not matter which option the user uses, the behaviour is the same.

Reply:
So the idea is to do this?

```erlang
erl_epmd_node_listen_port() ->
    PortParameter =
        case application:get_env(kernel, erl_epmd_node_listen_port) of
            {ok, Port} when is_integer(Port), Port >= 0 ->
                {ok, Port};
            {ok, Invalid} ->
                error({invalid_parameter_value, erl_epmd_node_listen_port, Invalid});
            undefined ->
                undefined
        end,
    PortArgument =
        try
            {ok, [[StringPort]]} = init:get_argument(erl_epmd_port),
            IntPort = list_to_integer(StringPort),
            {ok, IntPort}
        catch error:_ ->
            undefined
        end,
    case {PortParameter, PortArgument} of
        {undefined, undefined} -> undefined;
        {_, undefined} -> PortParameter;
        {undefined, _} -> PortArgument;
        _ -> error({invalid_configuration, "either -erl_epmd_port or kernel erl_epmd_node_listen_port should be specified, not both"})
    end.
```

One downside of doing this is that we read both places on every call. The current implementation parses the argument on the first call and puts it in the kernel configuration, as done for `-connect_all` in https://github.com/erlang/otp/commit/1d0fe6f71c320d2d1d19d9efd6dcba7b18bedca6. WDYT?

Reply:
I will get back to you on this one. I agree that it is not nice to read twice the same thing, and we also want to avoid having conflicting arguments, which is the reason for the crash

Reply:
Please, read it in both places even if it is cumbersome.
We would prefer a nice crash when there are conflicting options

Sorry for the delay, I forgot that I had to check this.

Reply:
Updated :)